File: schemas/sdg_prediction.py
from datetime import datetime
from pydantic import BaseModel

class SDGPredictionSchemaBase(BaseModel):
    prediction_id: int
    publication_id: int
    prediction_model: str
    sdg1: float
    sdg2: float
    sdg3: float
    sdg4: float
    sdg5: float
    sdg6: float
    sdg7: float
    sdg8: float
    sdg9: float
    sdg10: float
    sdg11: float
    sdg12: float
    sdg13: float
    sdg14: float
    sdg15: float
    sdg16: float
    sdg17: float
    predicted: bool
    last_predicted_goal: int

    model_config = {
        "from_attributes": True
    }


class SDGPredictionSchemaFull(SDGPredictionSchemaBase):
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend/types/sdgPrediction.ts
export interface SDGPredictionSchemaBase {
  predictionId: number;
  publicationId: number;
  predictionModel: string;
  sdg1: number;
  sdg2: number;
  sdg3: number;
  sdg4: number;
  sdg5: number;
  sdg6: number;
  sdg7: number;
  sdg8: number;
  sdg9: number;
  sdg10: number;
  sdg11: number;
  sdg12: number;
  sdg13: number;
  sdg14: number;
  sdg15: number;
  sdg16: number;
  sdg17: number;
  predicted: boolean;
  lastPredictedGoal: number;
}

export interface SDGPredictionSchemaFull extends SDGPredictionSchemaBase {
  createdAt: string;
  updatedAt: string;
}
